src/service-object.ts

```diff
@@ -329,16 +336,14 @@ class ServiceObject extends EventEmitter {
 
     // The `request` method may have been overridden to hold any special
     // behavior. Ensure we call the original `request` method.
-    this.request(reqOpts, (err: ApiError|null, resp?: r.Response) => {
-      if (err) {
-        callback(err, null, resp);
-        return;
-      }
-
-      self.metadata = resp as Metadata;
-
-      callback(null, self.metadata, resp);
-    });
+    this.request(reqOpts).then(
```

Comment:
Likewise, here and elsewhere, can we convert the containing method to async, and use `await`?

Reply:
In this case we can't.  `getMetadata` is a public function, and I don't want to make this change just yet.